test/e2e/specs/specs-playwright/shared-specs/privacy-testing.ts

```diff
@@ -0,0 +1,132 @@
+import {
+	DataHelper,
+	GutenbergEditorPage,
+	TestAccount,
+	envVariables,
+	PrivacyOptions,
+	PublishedPostPage,
+	EditorSettingsSidebarComponent,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+
+declare const browser: Browser;
+
+const postContent = DataHelper.getRandomPhrase();
+const postPassword = 'cat';
+
+/**
+ * Creates Page privacy tests.
+ *
+ * Accepts as parameter the intended page/post privacy option.
+ *
+ * @param param0 Keyed object parameter.
+ * @param {PrivacyOptions} param0.visibility Intended page/post privacy option.
+ */
+export function createPrivacyTests( { visibility }: { visibility: PrivacyOptions } ): void {
+	describe( DataHelper.createSuiteTitle( `Editor: Privacy` ), function () {
+		const accountName = envVariables.GUTENBERG_EDGE
+			? 'gutenbergSimpleSiteEdgeUser'
+			: 'simpleSitePersonalPlanUser';
+
+		let page: Page;
+		let url: string;
+		let gutenbergEditorPage: GutenbergEditorPage;
+		let editorSettingsSidebarComponent: EditorSettingsSidebarComponent;
+
+		describe( `Create a ${ visibility } post`, function () {
+			beforeAll( async function () {
+				page = await browser.newPage();
+
+				const testAccount = new TestAccount( accountName );
+				await testAccount.authenticate( page );
+			} );
+
+			it( 'Go to the new post page', async function () {
+				gutenbergEditorPage = new GutenbergEditorPage( page );
+				await gutenbergEditorPage.visit( 'page' );
+				await gutenbergEditorPage.selectPageDesign( 'blank' );
+			} );
+
+			it( 'Enter post title', async function () {
+				gutenbergEditorPage = new GutenbergEditorPage( page );
+				await gutenbergEditorPage.enterTitle(
+					`Privacy: ${ visibility } - ${ DataHelper.getTimestamp() }`
+				);
+			} );
+
+			it( 'Enter post content', async function () {
+				await gutenbergEditorPage.enterText( postContent );
+			} );
+
+			it( `Set post visibility to ${ visibility }`, async function () {
+				const frame = await gutenbergEditorPage.getEditorFrame();
+				editorSettingsSidebarComponent = new EditorSettingsSidebarComponent( frame, page );
+
+				await editorSettingsSidebarComponent.clickTab( 'Page' );
+				await editorSettingsSidebarComponent.setVisibility( visibility as PrivacyOptions );
+
+				if ( visibility === 'Password' ) {
+					await editorSettingsSidebarComponent.setPostPassword( postPassword );
+				}
+			} );
+
+			it( 'Publish post', async function () {
+				// Private articles are published immediately as the option is selected.
+				// In other words, for Private articles the publish action happened in previous step.
+				if ( visibility === 'Private' ) {
+					url = await gutenbergEditorPage.getPublishedURL();
+				} else {
+					url = await gutenbergEditorPage.publish();
+				}
+			} );
+		} );
+
+		describe( 'Validate post content', function () {
```

Comment:
After much thought and iteration, this is the way I've decided to split up the validation.

There are multiple behaviors that need to be taken into account. The following is a matrix of expected outcomes:

Public Page:
| user | visibility |
| not logged in | ✓  |
| non-posting user | ✓ |
| posting user | ✓ |

Password Page:
| user | visibility |
| not logged in | ✓  |
| non-posting user | ✓ |
| posting user | ✓ |

Private Page:
| user | visibility |
| not logged in | ☓  |
| non-posting user | ☓ |
| posting user | ✓ |

As noted, behavior of Public and Password pages are essentially the same - one has an additional step of entering the post password.

Behavior of Private pages are slightly more complex. Only the posting user is able to view the page exists.

Since the spec is designed to make a post of a particular type, then verify the expected outcome when viewed by each usertype, I felt that it was best to split up the validation based on the usertype in question.